### ISSUE TITLE
miRDeep2.pl

### ISSUE THREAD

**Issue by mpevs** [rajewsky-lab/mirdeep2#18]
I've run miRDeep2.pl successfully, but the output files have no values in the 'estimated probability that the miRNA candidate is a true positive" column, not even a '-' placeholder. Is this normal?

**Comment by mschilli87** [comment]
There are [many ways](https://github.com/rajewsky-lab/mirdeep2#examples) to 'run `miRDeep2.pl` successfully'. Without know *exactly* what you did, it will be tricky to help you by answering your question.

**Comment by mpevs** [comment]
Apologies for vagueness! In my command I supplied: 

- A FASTA file of reads
- A genome indexed with `bowtie-build`
- A file of reads mapped to the genome 

I ran it with no known miRNA files using `none none none` in the command (as in example 2 in the README). All other options were left as default. Here's the command in full (changed file names for clarity!). 

```
miRDeep2.pl ../readfile \
            ../genome \
            ../readfiles_mapped_to_genome\
            none none none  2> report.log
```

**Comment by mschilli87** [comment]
@mpevs:

1. Is this still an issue for you?
2. Can you reproduce it with the tutorial data?

@Drmirdeep: Any comment?

**Comment by dudemain** [comment]
Hi, I'm having the same issue. I've been running miRDeep2 for many precedent projects without ever having problems, but it seems that now, something's going wrong.

The output is attached and something seems to be bad with it.

The tutorial data went well as far as I can judge. Any help?


[result_14_12_2018_t_08_21_47IL.xlsx](https://github.com/rajewsky-lab/mirdeep2/files/2681443/result_14_12_2018_t_08_21_47IL.xlsx)

**Comment by mschilli87** [comment]
@dudemain: Can you share the exact version you use, how you installed it, and how you run miRDeep2?
Ideally, we'd have a set of small(!) input files to reproduce the issue. Otherwise, it will be hard to provide any help.

**Comment by dudemain** [comment]
@mschilli87: I'm using the last version if I'm right so miRDeep2.0.0.8, I installed it using the perl script install.pl, then I ran the run.tut.sh and everything looked fine. After this, I used an in inhouse script to run the whole thing. 

Right now I'm trying to run it on a subset of samples, so it's taking approximately 4h to run the whole miRDeep2.pl. 

What would be the best way for me to transfer the files to you? I could transfer all the required files, so the allsamples-aligned.fa, the fasta from the build I'm using, the allsamples_aligned.arf and the fasta files for the miRBase for mature, precursor and other species.

**Comment by dudemain** [comment]
@mschilli87 I just saw that miRDeep2.0.1.0 seems to be the last version, however when I clone from the github of miRDeep2, it is the miRDeep2.0.0.8 that is installed

**Comment by mschilli87** [comment]
@dudemain: This is an unrelated issue. You do have 2.0.1.0 installed, just the version was not bumped properly in the code:

https://github.com/rajewsky-lab/mirdeep2/blob/643a9cae329756c6d6c0ea61db98ec784e648d9c/src/miRDeep2.pl#L35

I'll fix this with the next release.

**Comment by dudemain** [comment]
@mschilli87: Some thought just popped in my mind, do you think the results could be due to the version of miRbase that I'm using? In the previous analyses I was always using the miRbase v21, but now, since the v22 is out, I used this one instead for the ongoing analysis. I doubt this could be the problem since it's only fasta file, however I think it might be worth something for you to know that.

**Comment by mschilli87** [comment]
@dudemain: I agree that miRBase v22 *should* not make a difference but I have not tried it myself.
If you can provide a *reproducible* example that works with miRBase v21 but not with v22, we'll obviously look into it so we can quickly release a new version that works with both.
If you can provide such an example (exact code & data) that doesn't work at all, I'll also do what I can to get it to work.
Without being able to see the error on my own system, I won't be able to troubleshoot your issue.

**Comment by dudemain** [comment]
Where can I provide you my files? Do you have any ftp where I could drop it?
I can send you the exact code that generated the weird results

**Comment by mschilli87** [comment]
Can you reproduce the issue by using only 10 of your reads? If so, you could post the FASTQ data here, and add links to where to download the miRBase reference, the genome reference etc and the code how to run miRDeep2.

**Comment by dudemain** [comment]
Not with only a few reads I fear! I'm trying to get the smaller file I could so you can reproduce something similar on your side. I need to target novel miRNA with different scores so you can see why at some point I got 0 +/- 0 (0 +/- 0%) while having novel miRNA having that score and on the other hand I'd like if it would be possible to point out why when the miRDeep2 score is decreasing, the estimated probability of the candidate being a true positive is also decreasing, that doesn't makes sense.
